src/org/labkey/test/util/AuditLogHelper.java

```diff
@@ -69,6 +76,116 @@ public DataRegionTable goToAuditEventView(String eventType)
+     * @throws IOException Can be thrown by the SelectRowsCommand.
+     * @throws CommandException Can be thrown by the SelectRowsCommand.
+     */
+    public static SelectRowsResponse getAuditLogsFromLKS(String projectName, String folderName, String auditEventName, List<String> columnNames,
+                                                         List<Filter> filters, @Nullable Integer maxRows) throws IOException, CommandException
+    {
+        SelectRowsCommand cmd = new SelectRowsCommand("auditLog", auditEventName);
+        cmd.setColumns(columnNames);
+        cmd.addFilter("ProjectId/DisplayName", projectName, Filter.Operator.EQUAL);
+        filters.forEach(cmd::addFilter);
+        if (maxRows != null)
+            cmd.setMaxRows(maxRows);
+
+        String formattedContainerPath = projectName;
+        if (!formattedContainerPath.startsWith("/"))
+            formattedContainerPath = "/" + formattedContainerPath;
+        if (!folderName.equalsIgnoreCase(projectName))
+            formattedContainerPath = formattedContainerPath + "/" + folderName;
+
+        return cmd.execute(WebTestHelper.getRemoteApiConnection(), formattedContainerPath);
+    }
+
+    /**
+     * Check the number of diffs in the audit event. This is a helper function to check the number of diffs in the
+     * newRecordMap for an audit entry. If a transactionId is provided, it will check all rows for that
+     * transactionId. If no transactionId is provided, it will check just the latest row.
+     */
+    public static void checkTimelineAuditEventDiffCount(String projectName, String folderName, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        checkTimelineAuditEventDiffCount(projectName, folderName, getAuditEventNameFromURL(), expectedDiffCounts);
+    }
+    public static void checkTimelineAuditEventDiffCount(String projectName, String folderName, String auditEventName, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        Integer maxRows = expectedDiffCounts.size();
+        List<Map<String, Object>> events = getAuditLogsFromLKS(projectName, folderName, auditEventName, List.of("NewRecordMap"), Collections.emptyList(), maxRows).getRows();
+        for (int i = 0; i < expectedDiffCounts.size(); i++)
+        {
+            Integer expectedDiffCount = expectedDiffCounts.get(i);
+            Map<String, Object> event = events.get(i);
+            String dataChangesStr = (String) event.get("NewRecordMap");
+            String[] dataChanges = dataChangesStr != null ? dataChangesStr.split("&") : new String[0];
+            // filter out SampleStateLabel as that is not a change, it is added for display purposes
+            dataChanges = Stream.of(dataChanges).filter(s -> !s.toLowerCase().startsWith("samplestatelabel=")).toArray(String[]::new);
+            BaseWebDriverTest.getCurrentTest().log("Audit record data changes diff count check: " + dataChangesStr);
```

Comment:
You can use `TestLogger.log` for logging from a static context.
```suggestion
            TestLogger.log("Audit record data changes diff count check: " + dataChangesStr);
```

Reply:
done